examples/main/main.cpp

```diff
@@ -865,9 +865,20 @@ int main(int argc, char ** argv) {
                 console::set_display(console::reset);
                 display = true;
 
-                // Add tokens to embd only if the input buffer is non-empty
-                // Entering a empty line lets the user pass control back
-                if (buffer.length() > 1) {
+                if (!buffer.length()) { // Ctrl+D on empty line exits
+                    LOG("EOF by user\n");
+                    break;
+                }
+                if (buffer[buffer.length() - 1] == '\n') {
```

Comment:
```suggestion
                }

                if (buffer[buffer.length() - 1] == '\n') {
```

Always add an empty line between 2 blocks of `if {}`